source/docs/software/commandbased/structuring-command-based-project.rst

```diff
@@ -1,239 +1,213 @@
+
+  .. tab:: C++ (Source)
+
+    .. remoteliteralinclude:: https://github.com/wpilibsuite/allwpilib/raw/master/wpilibcExamples/src/main/cpp/templates/commandbased/cpp/RobotContainer.cpp
+      :language: c++
+      :lines: 21-24
+      :linenos:
+      :lineno-start: 21
+
+Finally, the ``getAutonomousCommand()`` method provides a convenient way for users to send their selected autonomous command to the main ``Robot`` class (which needs access to it to schedule it when autonomous starts).
+
+Constants
+---------
+
+The ``Constants`` class (`Java <https://github.com/wpilibsuite/allwpilib/blob/master/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/templates/commandbased/Constants.java>`__, `C++ (Header) <https://github.com/wpilibsuite/allwpilib/blob/master/wpilibcExamples/src/main/cpp/templates/commandbased/include/Constants.h>`__) (in C++ this is not a class, but simply a header file in which several namespaces are defined) is where globally-accessible robot constants (such as speeds, unit conversion factors, PID gains, and sensor/motor ports) can be stored. It is recommended that users separate these constants into individual inner classes corresponding to subsystems or robot modes, to keep variable names shorter.
+
+In Java, all constants should be declared ``public static final`` so that they are globally accessible and cannot be changed.  In C++, all constants should be ``const``.
+
+For more illustrative examples of what a ``constants`` class should look like in practice, see those of the various command-based example projects:
+
+* FrisbeeBot (`Java <https://github.com/wpilibsuite/allwpilib/blob/master/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/frisbeebot/Constants.java>`__, `C++ <https://github.com/wpilibsuite/allwpilib/blob/master/wpilibcExamples/src/main/cpp/examples/Frisbeebot/include/Constants.h>`__)
+* GyroDriveCommands (`Java <https://github.com/wpilibsuite/allwpilib/blob/master/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/gyrodrivecommands/Constants.java>`__, `C++ <https://github.com/wpilibsuite/allwpilib/blob/master/wpilibcExamples/src/main/cpp/examples/GyroDriveCommands/include/Constants.h>`__)
+* Hatchbot (`Java <https://github.com/wpilibsuite/allwpilib/blob/master/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/hatchbottraditional/Constants.java>`__, `C++ <https://github.com/wpilibsuite/allwpilib/blob/master/wpilibcExamples/src/main/cpp/examples/HatchbotTraditional/include/Constants.h>`__)
 
-.. todo:: Link above to Constants.java and below to examples after merge
+In Java, it is recommended that the constants be used from other classes by statically importing the necessary inner class. An ``import static`` statement imports the static namespace of a class into the class in which you are working, so that any ``static`` constants can be referenced directly as if they had been defined in that class.  In C++, the same effect can be attained with ``using namespace``:
```

Comment:
I think this is a broad thing to recommend because this is not a library recommendation but a code style for Java as a whole.

Reply:
Not doing this makes command-based more verbose to use and isn't a general style requirement for Java, so we should mention best practices for our framework usage in context.